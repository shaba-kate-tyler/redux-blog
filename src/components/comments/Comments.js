import React from 'react';
import PropTypes from 'prop-types';
import Comment from './Comment';

function Comments({ comments }) {
  const commentsList = comments.map(comment => <li key={comment.id}>{<Comment {...comment}/>}</li>);
  return (
    <>
      <ul>
        {commentsList}
      </ul>
    </>
  );
}

Comments.propTypes = {
  comments: PropTypes.array.isRequired
};

export default Comments;
