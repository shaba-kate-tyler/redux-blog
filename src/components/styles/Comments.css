.Comments {
    background-color: rgb(244, 244, 244);
    margin: 2vh;
    align-items: center;
    list-style: none;
    padding: 2px;
}