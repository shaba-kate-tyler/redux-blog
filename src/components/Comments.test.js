import React from 'react';
import renderer from 'react-test-renderer';
import Comments from './Comments';

describe('Comments', () => {
  it('matches a snapshot', () => {
    const comments = [
      {
        postId: 1,
        id: 1,
        name: 'id labore ex et quam laborum',
        email: 'Eliseo@gardner.biz',
        body: 'laudantium enim quasi est quidem magnam voluptate ipsam eos tempora quo necessitatibus dolor quam autem quasi reiciendis et nam sapiente accusantium'
      },
      {
        postId: 1,
        id: 2,
        name: 'id labore ex et quam laborum',
        email: 'Eliseo@gardner.biz',
        body: 'laudantium enim quasi est quidem magnam voluptate ipsam eos tempora quo necessitatibus dolor quam autem quasi reiciendis et nam sapiente accusantium'
      },
      {
        postId: 1,
        id: 3,
        name: 'id labore ex et quam laborum',
        email: 'Eliseo@gardner.biz',
        body: 'laudantium enim quasi est quidem magnam voluptate ipsam eos tempora quo necessitatibus dolor quam autem quasi reiciendis et nam sapiente accusantium'
      }];
    const tree = renderer.create(
      <Comments comments={comments}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
