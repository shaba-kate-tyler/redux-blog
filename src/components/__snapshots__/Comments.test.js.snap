// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments matches a snapshot 1`] = `
<ul>
  <li>
    <p>
      laudantium enim quasi est quidem magnam voluptate ipsam eos tempora quo necessitatibus dolor quam autem quasi reiciendis et nam sapiente accusantium
    </p>
    <h4>
      -
      Eliseo@gardner.biz
    </h4>
  </li>
  <li>
    <p>
      laudantium enim quasi est quidem magnam voluptate ipsam eos tempora quo necessitatibus dolor quam autem quasi reiciendis et nam sapiente accusantium
    </p>
    <h4>
      -
      Eliseo@gardner.biz
    </h4>
  </li>
  <li>
    <p>
      laudantium enim quasi est quidem magnam voluptate ipsam eos tempora quo necessitatibus dolor quam autem quasi reiciendis et nam sapiente accusantium
    </p>
    <h4>
      -
      Eliseo@gardner.biz
    </h4>
  </li>
</ul>
`;
